### ISSUE TITLE
Allow int amp values for charging speeds other than reduced and maximum

### ISSUE THREAD

**Issue by stefanuc111** [lendy007/homeassistant-skodaconnect#49]
**Is your feature request related to a problem? Please describe.**

Hi I've a skoda citigo and I would like to select charging speed that I want but I've saw that currently it allow only "maximum" and "reduced" string values which map to 254 and 252, it would be nice to allaow also to take directly int values, infact I've tested by changing the code and if you pass a value from 5 to 32 (for skoda citigo) to the api it set directly the current to the correspond value

**Describe the solution you'd like**

Somethig like that:

```
    async def set_current(call):
        """Prepare data and modify for data call."""
        try:
            current = call.data.get('current')
            if current == "maximum":
                current = 254
            elif current == "reduced":
                current = 252
            _LOGGER.debug("Try to fetch object for VIN: %s" % call.data.get("vin", ""))
            vin = call.data.get("vin")
            car = connection.vehicle(vin)
            _LOGGER.debug(f"Found car: {car.nickname}")
            _LOGGER.debug(f"Set charger current to {call.data.get('current')} ({current})")
            await car.set_charger_current(current)
            async_dispatcher_send(hass, SIGNAL_STATE_UPDATED)
            return
        except Exception as error:
            _LOGGER.warning(f"Couldn't execute, error: {error}")
            async_dispatcher_send(hass, SIGNAL_STATE_UPDATED)
        raise Exception(f"Service call failed")    
```

```
SERVICE_SET_MAX_CURRENT_SCHEMA = vol.Schema(
    {
        vol.Required("vin"): cv.string,
        vol.Required("current"): vol.Any(vol.In(range(5, 33)), vol.In([ "maximum", "reduced"])),
    }
)
```

**Comment by Farfar** [comment]
> I've tested by changing the code and if you pass a value from 5 to 32 (for skoda citigo) to the api it set directly the current to the correspond value

Interesting! I have meant to test this someday but haven't gotten around to it.
I have examined how the app sets the value but haven't really found anything that I can use to determine what values are available for a given car. The same goes for the charge limit.

I will test of this also applies to other cars than the citigo and will then implement it.

**Comment by stefanuc111** [comment]
I thinks the apis should be almost the same as VW and in fact in the VW app you can actually set the ampere value instead of only "max" ore "reduced". I don't know who had the idea in skoda to put only two options in the app "max" or "reduced" without even explain at what power corresponds but it was a bad idea

**Comment by Farfar** [comment]
> I think the apis should be almost the same as VW and in fact in the VW app you can actually set the ampere value instead of only "max" ore "reduced". I don't know who had the idea in skoda to put only two options in the app "max" or "reduced" without even explain at what power corresponds but it was a bad idea

That's true for most Skodas. Not Enyaq iV though, it's running om Skoda native servers and different API calls.
My Superb iV is hosted on VW servers and use the same API calls as VW.
The new VW Passat GTE shares much with the Superb iV and it also only have maximum/reduced as options.

**Comment by stefanuc111** [comment]
I think the citigo should be on VW servers because it's pretty identic to the e-up

BTW nice work, congrats @Farfar and @lendy007 skodaconnect and homeassistant-skodaconnect are really a great project for skoda owners!

**Comment by Farfar** [comment]
Yes the Citigo is on VW servers. I don't know how this works for the e-up though, never had one and never debugged for one.



> BTW nice work, congrats @Farfar and @lendy007 skodaconnect and homeassistant-skodaconnect are really a great project for skoda owners!

Thanks!

**Comment by loeffelpan** [comment]
This would also improve the graph view.
At the moment the values „maximum“ and „reduced“ are displayed in an empty graph because the unit is ampere. But the values are not suitable.

![9980E146-9AAC-4169-B8F1-B0B73D43F22A](https://user-images.githubusercontent.com/34661317/129480696-f583e4a6-ef1d-40b0-8618-69032a7ccc7e.jpeg)

**Comment by Farfar** [comment]
I have tested and verified that integer values from 1 to 16 works for my Superb iV, it also correctly limits the maximum charging current to the correct value. For instance 8 ampere instead of being limited to reduced (6) and maximum (16).
This will be integrated in upcoming release 1.0.43-RC3

**Comment by loeffelpan** [comment]
Will that be a breaking change, or will recuded and maximum work additionally to integer values?

**Comment by Farfar** [comment]
> Will that be a breaking change, or will recuded and maximum work additionally to integer values?
> 
> I guess that setting an integer value will be implemented in the known service?

It's going to be a breaking change since I feel there's no point in supporting both when explicit numbers work better.

**Comment by Farfar** [comment]
> Will that be a breaking change, or will recuded and maximum work additionally to integer values?
> 
> I guess that setting an integer value will be implemented in the known service?

Scratch that. Seems like Enyaq uses maximum/minimum and not integer values so I will include both options.

**Comment by loeffelpan** [comment]
Are u sure. Never saw it by myself, but I‘ve heard that Enyaq has more than two charging speeds?

**Comment by Farfar** [comment]
> Are u sure? Never saw it by myself, but I‘ve heard that Enyaq has more than two charging speeds.

It might have many more charging speeds but in the app there's only the same two options: maximum and reduced.
